### ISSUE TITLE
dmd2.067.1 linking error

### ISSUE THREAD

**Issue by chalucha** [rejectedsoftware/vibe.d#1097]
Just following the:
```
dub init <name> vibe.d
cd <name>
dub
```

ends up with:
```
Building vibe-d 0.7.23 configuration "libevent", build type debug.
Running dmd...
Building vibetest ~master configuration "application", build type debug.
Compiling using dmd...
Linking...
.dub/build/application-debug-linux.posix-x86_64-dmd_2067-5FA2C7F44CBC2A9060144DDC4B388A16/vibetest.o: In function `std.digest.sha.SHA!(512, 160).SHA._sharedStaticCtor3347()':
/opt/dmd-2.067/import/std/digest/sha.d:224: undefined reference to `std.internal.digest.sha_SSSE3.transformSSSE3(uint[5]*, const(ubyte[64])*)'
collect2: error: ld returned 1 exit status
--- errorlevel 1
FAIL .dub/build/application-debug-linux.posix-x86_64-dmd_2067-5FA2C7F44CBC2A9060144DDC4B388A16/ vibetest executable
Error executing command build:
dmd failed with exit code 1.
```

Weird is that with dub -b release it compiles fine.
LDC2 is also working.

I'm on gentoo linux x86_64.
Not sure if this is vibe.d or dmd related though.

**Comment by etcimon** [comment]
Did you compile DMD/druntime/phobos by yourself? What linker are you using (ld -v)?

**Comment by chalucha** [comment]
ld -v output:
```GNU ld (Gentoo 2.25 p1.0) 2.25```

dmd is called from /opt/dmd-2.067/bin/dmd - installed from dlang gentoo overlay

I have actually a dmd/druntime/phobos in /usr/src/d but never installed it from there. Even if I remove it and reinstall dmd from overlay, it stays the same.

**Comment by chalucha** [comment]
I narrowed it to serializeToJson to counter called within writeJsonBody.
So actually route is found, but no json writen to response.

I've made simple test app:
```D
long length = 0;
auto counter = RangeCounter(&length);
serializeToJson(counter, [MyType(1, "testa"), MyType(2, "testb")]);
return 0;
```
which never returns 0 with dmd -release -O -inline
I dustmited it to this: http://pastebin.com/bsSjPGN3

Not sure if it's the same problem, but again it works in debug and not working in -release -O -inline
Hope it helps.
